### PR TITLE
various: add zaps

### PR DESCRIPTION
**Pull request by krehel**
Add zaps

* 010-editor
* sapmachine-jdk
* warsow
* zeronet

Update zap

* arduino-ide
* clone-hero
* whatsize

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

### CODE REVIEW

**Comment by krehel** [comment]
Hi team - updated multiple zaps.  Let me know if the preference is to break these into their own pulls and I'll resubmit separately.

Thanks

File: Casks/whatsize.rb [hunk 1]
```diff
@@ -20,13 +20,13 @@
             launchctl: "com.id-design.v#{version.major}.whatsizehelper"
 
   zap trash: [
-    "/Library/LaunchDaemons/com.id-design.v7.whatsizehelper.plist",
-    "/Library/PrivilegedHelperTools/com.id-design.v7.whatsizehelper",
+    "/Library/LaunchDaemons/com.id-design.v#{version.major}.whatsizehelper.plist",
+    "/Library/PrivilegedHelperTools/com.id-design.v#{version.major}.whatsizehelper",
```
Comment (razvanazamfirei) [inline Casks/whatsize.rb:24]: Are these files actually getting deleted? Some of them may need to be added to `zap delete:`

File: Casks/arduino-ide.rb [hunk 2]
```diff
@@ -22,5 +22,6 @@
   zap trash: [
     "~/.arduinoIDE",
     "~/Library/Application Support/arduino-ide",
+    "~/Library/Saved Application State/cc.arduino.IDE2.savedState",
```
Comment (razvanazamfirei) [inline Casks/arduino-ide.rb:25]:
```suggestion
    "~/Library/Saved Application State/cc.arduino.IDE#{version.major}.savedState",
```

**Review by razvanazamfirei — COMMENTED** [review]
Thanks for submitting this and all the hard work! 😄  We generally don't encourage submitting multiple casks in a PR, but if the number is kept reasonably low and the casks are not problematic, it should be fine. If there's a possibility that CI would fail, it should be in a separate pull request.  A few comments:

File: Casks/010-editor.rb [hunk 3]
```diff
@@ -13,4 +13,11 @@
   end
 
   app "010 Editor.app"
+
+  zap trash: [
+    "~/.config/SweetScape",
+    "~/Documents/SweetScape",
```
Comment (bevanjkay) [inline Casks/010-editor.rb:19]: Does this folder contain user-created files. If so we should not trash it but could add it to `zap rmdir` do that it is removed if empty.

Comment (krehel) [reply Casks/010-editor.rb:19]: @bevanjkay, I thought perhaps the same thing.  Both are populated with application files, but I am happy to move both to an rmdir for safety.

File: Casks/arduino-ide.rb (hunk 2 above)
Comment (krehel) [reply Casks/arduino-ide.rb:25]: Thanks @razvanazamfirei,

Submitting changes.  Little too excited with the zap on whatsize, we can remove those files you noted as they are removed as part of uninstall.

Comment (krehel) [reply Casks/arduino-ide.rb]: Pushed commit for this!

File: Casks/whatsize.rb (hunk 1 above)
Comment (krehel) [reply Casks/whatsize.rb]: Pushed a fix to remove these files

File: Casks/010-editor.rb (hunk 3 above)
Comment (krehel) [reply Casks/010-editor.rb:19]: To add more context...

By default -

~/.config/Sweetscape contains an editor .ini file
~/Documents/SweetScape contains binary template files and 010-editor script files.  Given users could make their own and place into at least the ~/Documents folder, I agree we should move at least that to rmdir.

**Review by razvanazamfirei — APPROVED** [review]

**Comment by krehel** [comment]
@razvanazamfirei and @bevanjkay - thank you both for your patience and guidance.  I'll continue to strive to improve the quality of my commits.

Appreciate it greatly!

**Comment by razvanazamfirei** [comment]
@krehel, you're doing great! :)

**Pull request merged**